calour/dsfdr.py

```diff
@@ -238,6 +238,24 @@ def dsfdr(data, labels, transform_type='rankdata', method='meandiff',
         labels = labels - np.mean(labels)
         tstat = np.dot(data, labels)
         t = np.abs(tstat)
+
+        # calculate the normalized test statistic
+        stdval = np.std(data, axis=1).reshape([data.shape[0], 1])
+        # to fix problem with 0 std divide by zero (since we permute it's ok)
+        # note we don't remove from mutiple hypothesis - could be done better
+        stdval[stdval == 0] = 1
+        tdata = data / np.repeat(stdval, data.shape[1], axis=1)
+        meanval = np.mean(tdata, axis=1).reshape([tdata.shape[0], 1])
+        tdata = tdata - np.repeat(meanval, tdata.shape[1], axis=1)
+        meanval = np.mean(data, axis=1).reshape([data.shape[0], 1])
+        tdata = tdata - np.repeat(meanval, tdata.shape[1], axis=1)
+
+        tlabels = labels / np.std(labels)
+        # fix for n since we multiply without normalizing for n
+        tlabels = tlabels / len(tlabels)
+        tlabels = tlabels - np.mean(tlabels)
+        tstat = np.dot(tdata, tlabels)
+
```

Comment:
is this duplicate of the code in the last PR?

Reply:
i think so... somehow the PRs got mixed up?